src/stream_engine_base.cpp

```diff
@@ -689,6 +694,15 @@ void zmq::stream_engine_base_t::error (error_reason_t reason_)
             || _mechanism->status () == mechanism_t::handshaking)) {
         const int err = errno;
         _socket->event_handshake_failed_no_detail (_endpoint_uri_pair, err);
+#ifdef ZMQ_BUILD_DRAFT_API
```

Comment:
It's not the ifdef that's missing - the new option needs to be double-defined in src/zmq_drafts.h

Reply:
Thanks!

I  *think* I get it:  zmq_draft.h is there to #define constants that would otherwise be undefined in a non-draft build.  Doing that helps to minimize the #ifdef's scattered through the code.  Curious if that's documented anywhere -- I wasn't able to find it .

Reply:
Yes that's it - I don't it we've written it anywhere